### PR TITLE
Bug 1627027 - Use nightly Fenix variant.

### PR DESCRIPTION
**Pull request by gmierz**
This patch switches the fenix build variant from performance-test to nightly.

### CODE REVIEW

**Comment by gmierz** [comment]
@JohanLorenzo this is a test with the nightly build variant. There are failures here but after the installation stage - the remaining failures need fixes in mozilla-central.

File: taskcluster/ci/browsertime/kind.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ job-defaults:
                     subject: '[{product_name}] Raptor-Browsertime job "{task_name}" failed'
                     to-addresses: [perftest-alerts@mozilla.com]
             default: {}
-    run-on-tasks-for: []
+    run-on-tasks-for: [github-pull-request]
```
Comment (JohanLorenzo) [inline taskcluster/ci/browsertime/kind.yml:39]: Nit: I guess we don't want to run these tests on all PRs. I'm sure you know, I'm just making it explicit: Let's remove change before landing 🙂

File: taskcluster/ci/raptor/kind.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ job-defaults:
                     subject: '[{product_name}] Raptor job "{task_name}" failed'
                     to-addresses: [perftest-alerts@mozilla.com]
             default: {}
-    run-on-tasks-for: []
+    run-on-tasks-for: [github-pull-request]
```
Comment (JohanLorenzo) [inline taskcluster/ci/raptor/kind.yml:37]: Nit: same comment

**Comment by gmierz** [comment]
> This looks good to me from a taskgraph perspective! Let's make sure tests are green before landing. I remember we needed to create the `performance-test` flavor because regular builds weren't correctly instrumented. If this is not an issue anymore today, then I'm happy to retire this build flavor smiley

The failures were because there was no existing conditioned profile for the fenix-nightly build variant (we only had performance-test available). It's available now from this try run: https://treeherder.mozilla.org/#/jobs?repo=try&selectedTaskRun=frxZKZ0ER9y_JZIRrKraBg-0&tier=1%2C2%2C3&revision=2085d0e41443634de2c0397694516660351f8390

So I force-updated the patch with a rebase and we'll see if that solves it all here. I have a feeling I'll have to increase the max-run-time for YTP since it seems to be taking longer in my try runs.

From what I know, the only difference between performance-test and nightly now is that performance-test is a debuggable build (this was for something other than raptor IIIRC - or maybe it had web-extension stuff enabled for it while the other variants didn't). In either case, this isn't representative of what a user may experience which is the main reason why we're making the switch. @mcomella would have more information about these differences. I'm not sure if others might be using this build for anything, so you'll want to confirm with him or someone else who's more familiar with it.

**Comment by gmierz** [comment]
Failures were caused because the visual-metrics script needs to be updated to include the new similarity metrics for live-site stability testing.

**Comment by gmierz** [comment]
I've re-requested a review since there's some extra change being added here.

**Comment by gmierz** [comment]
There may be a high-frequency intermittent failure in `browsertime-tp6m-1-cold-nightly-armeabi-v7a` - it's a known failure.

File: taskcluster/docker/visual-metrics/Dockerfile [hunk 3]
```diff
@@ -20,10 +20,11 @@ WORKDIR /builds/worker
 USER worker:worker
 
 COPY requirements.txt /builds/worker/requirements.txt
-RUN pip3 install setuptools==46.0.0
+RUN pip3 install setuptools==46.1.3
```
[the remark this answers is not in the document]
Comment (gmierz) [reply taskcluster/docker/visual-metrics/Dockerfile:23]: I don't think there is anything preventing that change but I'd like to keep the dockerfile as similar as possible to the mozilla-central version. Is that fine for you? I'm happy to make the change if you'd like and do the same in m-c if there are no issues.

**Comment by gmierz** [comment]
> I'm rubber-stamping the `similarity` part which I assume was copied from central.

Yes, it provides a couple new metrics we're testing out in production: https://bugzilla.mozilla.org/show_bug.cgi?id=1634406

If the similarity calculation fails, it's ignored and won't impact the standard visual metrics calculations.

File: taskcluster/docker/visual-metrics/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (gmierz) [reply taskcluster/docker/visual-metrics/Dockerfile]: Fixed.

**Comment by gmierz** [comment]
@JohanLorenzo if you're happy with this (including the latest setuptools change), would you be able to land this when you get a chance please? :)

**Comment by gmierz** [comment]
Browsertime tasks are perma-failing in the fenix tree. I've added a patch to fix them (the aritfact suffix changed).

**Comment by gmierz** [comment]
Thanks for the merge and reviews @JohanLorenzo! :)

**Pull request merged**